doc/v5.0/man3/MPI_Win_flush_local.3
.\" -*- nroff -*-
.\" Copyright 2014 Los Alamos National Security, LLC. All rights reserved.
.\" Copyright 2010 Cisco Systems, Inc.  All rights reserved.
.\" Copyright 2007-2008 Sun Microsystems, Inc.
.\" Copyright (c) 1996 Thinking Machines Corporation
.\" $COPYRIGHT$
.TH MPI_Win_flush_local 3 "Sep 30, 2021" "5.0.0rc1" "Open MPI"
.SH NAME
\fBMPI_Win_flush_local\fP, \fBMPI_Win_flush_local_all\fP \- Complete all outstanding RMA operations at both the origin

.SH SYNTAX
.ft R
.SH C Syntax
.nf
#include <mpi.h>
int MPI_Win_flush_local (int \fIrank\fP, MPI_Win \fIwin\fP)

int MPI_Win_flush_local_all (MPI_Win \fIwin\fP)

.fi
.SH Fortran Syntax
.nf
USE MPI
! or the older form: INCLUDE 'mpif.h'
MPI_WIN_FLUSH_LOCAL(\fIRANK, WIN, IERROR\fP)
	INTEGER \fIRANK, WIN, IERROR\fP

MPI_WIN_FLUSH_LOCAL_ALL(\fIWIN, IERROR\fP)
	INTEGER \fIWIN, IERROR\fP

.fi
.SH Fortran 2008 Syntax
.nf
USE mpi_f08
MPI_Win_flush_local(\fIrank\fP, \fIwin\fP, \fIierror\fP)
	INTEGER, INTENT(IN) :: \fIrank\fP
	TYPE(MPI_Win), INTENT(IN) :: \fIwin\fP
	INTEGER, OPTIONAL, INTENT(OUT) :: \fIierror\fP

MPI_Win_flush_local_all(\fIwin\fP, \fIierror\fP)
	TYPE(MPI_Win), INTENT(IN) :: \fIwin\fP
	INTEGER, OPTIONAL, INTENT(OUT) :: \fIierror\fP

.fi
.SH INPUT PARAMETERS
.ft R
.TP 1i
rank
Rank of window (nonnegative integer).
.TP 1i
win
Window object (handle).

.SH OUTPUT PARAMETER
.ft R
.TP 1i
IERROR
Fortran only: Error status (integer).

.SH DESCRIPTION
.ft R
\fBMPI_Win_flush_local\fP locally completes at the origin all outstanding RMA operations initiated by the calling process to the target process specified by rank on the specified window. For example, after this routine completes, the user may reuse any buffers provided to put, get, or accumulate operations. \fBMPI_Win_flush_local_all\fP locally completes at the origin all outstanding RMA operations to all targets.
.sp
Can only be called from within a passive target epoch.

.SH ERRORS
Almost all MPI routines return an error value; C routines as the value of the function and Fortran routines in the last argument.
.sp
Before the error value is returned, the current MPI error handler is
called. By default, this error handler aborts the MPI job, except for I/O function errors. The error handler may be changed with \fBMPI_Comm_set_errhandler\fP; the predefined error handler MPI_ERRORS_RETURN may be used to cause error values to be returned. Note that MPI does not guarantee that an MPI program can continue past an error.

.SH SEE ALSO
MPI_Win_flush
MPI_Win_lock
MPI_Win_lock_all
.br


doc/v2.1/man3/MPI_Win_flush_local.3.php
<?php
$topdir = "../../..";
$title = "MPI_Win_flush_local(3) man page (version 2.1.6)";
$meta_desc = "Open MPI v2.1.6 man page: MPI_WIN_FLUSH_LOCAL(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
      <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_Win_flush_local</b>, <b><a href="../man3/MPI_Win_flush_local_all.3.php">MPI_Win_flush_local_all</a></b> - Complete all outstanding

<p>RMA operations at both the origin
<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<h2><a name='sect2' href='#toc2'>C Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_Win_flush_local (int rank, MPI_Win win)
int <a href="../man3/MPI_Win_flush_local_all.3.php">MPI_Win_flush_local_all</a> (MPI_Win win)
</pre>
<h2><a name='sect3' href='#toc3'>Fortran Syntax</a></h2>
<br>
<pre>INCLUDE &rsquo;mpif.h&rsquo;
MPI_WIN_FLUSH_LOCAL(RANK, WIN, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER RANK, WIN, IERROR
<a href="../man3/MPI_Win_flush_local_all.3.php">MPI_WIN_FLUSH_LOCAL_ALL</a>(WIN, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER WIN, IERROR
</pre>
<h2><a name='sect4' href='#toc4'>Input Parameters</a></h2>

<dl>

<dt>rank </dt>
<dd>Rank of window (nonnegative integer). </dd>

<dt>win </dt>
<dd>Window object
(handle).
<p> </dd>
</dl>

<h2><a name='sect5' href='#toc5'>Output Parameter</a></h2>

<dl>

<dt>IERROR </dt>
<dd>Fortran only: Error status (integer).
<p>
</dd>
</dl>

<h2><a name='sect6' href='#toc6'>Description</a></h2>
<b>MPI_Win_flush_local</b> locally completes at the origin all outstanding
RMA operations initiated by the calling process to the target process specified
by rank on the specified window. For example, after this routine completes,
the user may reuse any buffers provided to put, get, or accumulate operations.
<b><a href="../man3/MPI_Win_flush_local_all.3.php">MPI_Win_flush_local_all</a></b> locally completes at the origin all outstanding
RMA operations to all targets. <p>
Can only be called from within a passive
target epoch.
<p>
<h2><a name='sect7' href='#toc7'>Errors</a></h2>
Almost all MPI routines return an error value; C routines
as the value of the function and Fortran routines in the last argument.
<p>
Before the error value is returned, the current MPI error handler is called.
By default, this error handler aborts the MPI job, except for I/O function
errors. The error handler may be changed with <b><a href="../man3/MPI_Comm_set_errhandler.3.php">MPI_Comm_set_errhandler</a></b>; the
predefined error handler MPI_ERRORS_RETURN may be used to cause error values
to be returned. Note that MPI does not guarantee that an MPI program can
continue past an error.
<p>
<h2><a name='sect8' href='#toc8'>See Also</a></h2>
<a href="../man3/MPI_Win_flush.3.php">MPI_Win_flush</a> <a href="../man3/MPI_Win_lock.3.php">MPI_Win_lock</a> <a href="../man3/MPI_Win_lock_all.3.php">MPI_Win_lock_all</a>
<br>
<p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>C Syntax</a></li>
<li><a name='toc3' href='#sect3'>Fortran Syntax</a></li>
<li><a name='toc4' href='#sect4'>Input Parameters</a></li>
<li><a name='toc5' href='#sect5'>Output Parameter</a></li>
<li><a name='toc6' href='#sect6'>Description</a></li>
<li><a name='toc7' href='#sect7'>Errors</a></li>
<li><a name='toc8' href='#sect8'>See Also</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
